### PR TITLE
synergy: 1.13.1.41 -> 1.14.1.32

### PR DESCRIPTION
**Pull request by talyz**
###### Motivation for this change
Upgrade `synergy` to the latest stable release. Replaces #144267.

While it builds on darwin and the CLI works fine, I can't get the GUI to work on Big Sur - the program menu shows up, but is unresponsive and no main window ever shows up. This doesn't seem to be caused by the upgrade, however, since the older versions of the package don't work either. I also can't find any Qt application in `nixpkgs` that works on Big Sur - `keepassxc` and `bambootracker` exhibit the same behavior. Edit: This issue is addressed in #145473.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/synergy/default.nix [hunk 1]
```diff
@@ -22,20 +50,42 @@ stdenv.mkDerivation rec {
...
   '';
 
-  cmakeFlags = lib.optional (!withGUI) "-DSYNERGY_BUILD_LEGACY_GUI=OFF";
+  cmakeFlags = lib.optionals (!withGUI) [
+    "-DSYNERGY_BUILD_LEGACY_GUI=OFF"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.09"
+  ];
+  NIX_CFLAGS_COMPILE =
+    lib.optionalString stdenv.isDarwin "-Wno-inconsistent-missing-override";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/synergy/default.nix:59]:
```suggestion
  ];

  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-Wno-inconsistent-missing-override";
```

File: pkgs/applications/misc/synergy/default.nix [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
   src = fetchFromGitHub {
     owner = "symless";
     repo = "synergy-core";
     rev = "${version}-stable";
     fetchSubmodules = true;
-    sha256 = "1phg0szc9g018zxs5wbys4drzq1cdhyzajfg45l6a3fmi6qdi1kw";
+    sha256 = "123p75rm22vb3prw1igh0yii2y4bvv7r18iykfvmnr41hh4w7z2p";
   };
 
   patches = lib.optional stdenv.isDarwin ./macos_build_fix.patch;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/synergy/default.nix:46]: Can we apply this patch unconditionally to prevent bitrot?

Comment (talyz) [reply pkgs/applications/misc/synergy/default.nix:46]: Sure, the patch shouldn't affect the build on Linux/NixOS.

File: pkgs/applications/misc/synergy/macos_build_fix.patch [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
+--- a/src/gui/src/OSXHelpers.mm
++++ b/src/gui/src/OSXHelpers.mm
+@@ -20,10 +20,6 @@
+ #import <Foundation/Foundation.h>
+ #import <CoreData/CoreData.h>
+ #import <Cocoa/Cocoa.h>
+-#import <UserNotifications/UNNotification.h>
+-#import <UserNotifications/UNUserNotificationCenter.h>
+-#import <UserNotifications/UNNotificationContent.h>
+-#import <UserNotifications/UNNotificationTrigger.h>
```
Comment (jonringer) [inline pkgs/applications/misc/synergy/macos_build_fix.patch:26]: Feel like this isn't nixpkgs specific, I would like to see it upstreamed if it makes sense.

But this doesn't have to be a blocker for this PR.

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

```
https://github.com/NixOS/nixpkgs/pull/145066

3 packages built:
quicksynergy synergy synergyWithoutGUI
```

File: pkgs/applications/misc/synergy/macos_build_fix.patch (hunk 3 above)
Comment (talyz) [reply pkgs/applications/misc/synergy/macos_build_fix.patch:26]: Yep, I'll send in a patch upstream. It just needs to be made conditional, like the code which uses the imported headers already is.

**Comment by talyz** [comment]
#145473 solves the Big Sur issue.

**Pull request merged**